learningresources/models.py

```diff
@@ -79,14 +82,38 @@ def has_resources(self):
 
     @transaction.atomic
     def save(self, *args, **kwargs):
-        """Handle slugs."""
+        """Handle slugs and groups"""
+        is_create = False
+        is_update = False
         if self.id is None or self.name != get_object_or_404(
                 Repository, id=self.id).name:
+            # if it is an update of the repository, need the old slug
+            if self.id is not None and \
+                    self.name != get_object_or_404(Repository,
```

Comment:
But you've already checked `self.name != get_object_or_404(Repository, id=self.id).name` a few lines above